__tests__/app.test.js

```diff
@@ -157,3 +163,52 @@ describe("GET/api/articles/:article_id/comments", () => {
     });
   });
 });
+describe("POST/api/articles/:article_id/comments", () => {
+  test("should return the comment from the user with the correct properties and a status 201", () => {
+    const inputComment = {
+      username: "lurker",
+      body: "this is a new comment",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(inputComment)
+      .expect(201)
+      .then((response) => {
+       
+        const newComment = response.body.comment;
+        expect(newComment).toMatchObject({
+          body: expect.any(String),
+          votes: expect.any(Number),
+          author: expect.any(String),
+          article_id: expect.any(Number),
+          created_at: expect.any(String),
+        });
+      });
+  });
+  describe("error handling for post comment", () => {
+    test("should response with Status 400 if the body is missing", () => {
+      const inputComment = {
+        username: "lurker",
+      };
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send(inputComment)
+        .expect(400)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe("Bad request");
+        });
+    });
+    test("should response with Status 400 if the username is missing", () => {
+      const inputComment = {
+        body: "body without username",
+      };
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send(inputComment)
+        .expect(400)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe("Bad request");
+        });
+    });
+  });
```

Comment:
We should add a test for a invalid id, a non existent id and for when we try to post but the username does not exist, which would be a 404